mlir/lib/Dialect/MIOpen/Transforms/AffixTuningParameters.cpp

```diff
@@ -98,32 +100,55 @@ void AffixTuningParameters::runOnFunction() {
   func.walk([&](miopen::GridwiseGemmOp op) {
     OpBuilder b(op.getContext());
 
-    // TBD. Compute tuning parameters from actual logic.
-    op.setAttr("block_size", b.getI32IntegerAttr(256));
+    ConvolutionContext convContext = populateConvContext(op);
+    InitParamsNonXDL validParams{0, 0, 0, 0, 0, 0};
+    GemmSize gemmSize;
+    DerivedParams gemmADerivedParam;
+    DerivedParams gemmBDerivedParam;
+    int64_t gemmCDstPerWrite;
+    int64_t gridSize;
 
-    op.setAttr("m_per_block", b.getI32IntegerAttr(128));
-    op.setAttr("n_per_block", b.getI32IntegerAttr(128));
-    op.setAttr("k_per_block", b.getI32IntegerAttr(8));
+    PopulateParams populateParams;
+    populateParams.paramsFromCtx(convContext, validParams, gemmSize,
```

Comment:
@jerryyin could you document the differences between:

- ConvolutionContext
- InitParamsNonXDL
- GemmSize
- DerivedParams
- PopulateParams

Reply:
`PopulateParams::paramsFromCtx` should better returns a `LogicalResult` so error processing logic should be added here.

Reply:
Will document different subroutines and data strctures in a header file.

How do we plan to handle the error when a `LogicalResult` is returned? I kind of like the existing way of using `llvm_unreachable` to yell at the user and abort. After all, if the routine is not able to decide at a set of parameter, more than likely it is due to the user has given a wrong (incoherent) set of parameters to start with.

Reply:
I think I find it, the proper way to give up is to use `signalPassFailure()` when `runOnFunction()`. Will do.